tools/FakeItEasy.Deploy/Program.cs

```diff
@@ -63,9 +64,29 @@ public static async Task Main(string[] args)
                 await UploadPackageToNuGetAsync(file, nugetServerUrl, nugetApiKey);
             }
 
+            var issueNumbersInCurrentRelease = ReleaseHelpers.GetIssueNumbersReferencedFromReleases(new[] { release });
+            var preReleases = GetPreReleasesContributingToThisRelease(release, releases);
+            var issueNumbersInPreReleases = ReleaseHelpers.GetIssueNumbersReferencedFromReleases(preReleases);
+            var newIssueNumbers = issueNumbersInCurrentRelease.Except(issueNumbersInPreReleases);
+
+            Console.WriteLine($"Adding 'released as part of' notes to {newIssueNumbers.Count()} issues");
+            var commentText = $"This change has been released as part of [{repoName} {releaseName}](https://github.com/{repoOwner}/{repoName}/releases/tag/{releaseName}).";
+            await Task.WhenAll(newIssueNumbers.Select(n => gitHubClient.Issue.Comment.Create(repoOwner, repoName, n, commentText)));
+
             Console.WriteLine("Finished deploying");
         }
 
+        private static IEnumerable<Release> GetPreReleasesContributingToThisRelease(Release release, IReadOnlyList<Release> releases)
+        {
+            if (release.Prerelease)
+            {
+                return Enumerable.Empty<Release>();
+            }
+
+            string baseName = release.Name.Split('-')[0];
+            return releases.Where(r => r.Prerelease && r.Name.StartsWith(baseName));

```

Comment:
I don't think this filter is sufficient. If the release is `5.1.1`, it will match `5.1.10-beta.1`.

Of course, under normal circumstances 5.1.10-xxx wouldn't exist... But I think it wouldn't hurt to include the dash in `basename`, i.e. match only issues starting with `5.1.1-`.

Reply:
Yeah, I thought of this last night. And was too lazy. In addition to thinking that 5.1.10-xxx wouldn't exist, even if it did, it just means we're remove issues attributed to it from the thanking mechanism for 5.1.1, which... is fine?

But I'm happy to change.